CMakeLists.txt

```diff
@@ -1,10 +1,275 @@
 # Native code side of Thrive
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.13)
+
+set(CMAKE_CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo;Distribution" CACHE STRING "" FORCE)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Force x64 for Windows before project() call
+if(WIN32)
+    set(CMAKE_GENERATOR_PLATFORM x64)
+endif()
 
 project(Thrive)
 
 # If you want to get compile commands run cmake with
 # "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+# Options
+# Building either the faster variant with AVX or without for older CPU support
+
+# CPU core optimization 
+include(ProcessorCount)
+ProcessorCount(CPU_CORES)
+if(CPU_CORES EQUAL 0)
+    set(CPU_CORES 1)
+elseif(CPU_CORES GREATER 4)
+    math(EXPR CPU_CORES "${CPU_CORES} - 1")
+endif()
+
+# Function to configure unity build size based on CPU cores
+function(configure_cpu_optimized_unity_build target)
+    if(WIN32 AND MSVC)
+        if(CPU_CORES LESS 4)
+            set(UNITY_BATCH_SIZE 10)
+        elseif(CPU_CORES LESS 8)
+            set(UNITY_BATCH_SIZE 20)
+        else()
+            set(UNITY_BATCH_SIZE 30)
+        endif()
+
+        set_target_properties(${target} PROPERTIES
+            UNITY_BUILD ON
+            UNITY_BUILD_MODE GROUP
+            UNITY_BUILD_BATCH_SIZE ${UNITY_BATCH_SIZE}
+            VS_GLOBAL_EnableUnitySupport "true"
+            VS_GLOBAL_UseUnitySupport "true"
+            VS_GLOBAL_UnitySupport "true"
+        )
+    endif()
+endfunction()
+
+# Windows detection function
+function(configure_windows_target target)
+    if(NOT WIN32 OR NOT MSVC)
+        return()
+    endif()
+
+    string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+# These are the survivors that earned their right to silence
+    target_compile_options(${target} PRIVATE
+        /MP${CPU_CORES}
+        /bigobj
+        /GR 
+        /EHsc
+        /nologo
+        /WX-
+        /Zc:preprocessor
+        /Zc:__cplusplus
+        /Zc:externConstexpr
+        /Zc:throwingNew
+        /Zc:rvalueCast
+        /FI"cstddef"
+        /FI"cstdint"
+        /FI"atomic"
+        $<$<CONFIG:Debug>:/MDd /Od /Ob1 /RTC1>
+        $<$<NOT:$<CONFIG:Debug>>:/MD /O2 /Ob2 /DNDEBUG>
+        # Critical warning suppressions
+        /wd4005  # Macro redefinition
+        /wd4273  # Inconsistent dll linkage
+        /wd4141  # 'inline' used more than once
+        # Other warning disables
+        /wd4068  # Unknown pragma
```

Comment:
If the thrive code doesn't trigger any of these warnings it would be, much much better to not include these. In general you have made a ton of changes here, whereas I'd prefer to *only* have the things that are truly, truly needed and not just list everything just in case at it makes it super hard to see what actually is needed.

Reply:
the problem isn't in the thrive code pal, the problem is in the thousands of files from the godot library which tended to give a lot of warnings and sometimes errors making the compile task quite tedious

Reply:
Then make it so that the godot library gets added these as public compile flags, that's the modern way to handle this in cmake. I see quite a lot of cmake usage where the old way to do things is used instead of the modern and more clean cmake way.